### DISCUSSION TITLE
Modal

### DISCUSSION THREAD

**Discussion by PARKER-X** [GeriLife/companionship-care#203] [Q&A]
Can u explain me this code how the data is saved and also how can we see companion name
if i use {{activity.remaining_eligible_companions}} they so me list
if i use {{activity.id}}  i show only one number 

and 
what is the use of {{activity.id}}  in data bs target

 
                {% if user in circle.organizers %}
                    {% if activity.remaining_eligible_companions %}
                        <button
                            type="button"
                            title="{% translate 'Add' %}"
                            style="border: none;"
                            class="badge rounded-pill bg-default text-dark"
                            data-bs-toggle="modal"
                            data-bs-target="#add-activity-participant-modal-{{ activity.id }}"
                        >
                            <i class="bi bi-person"></i>
                            {% translate 'Add' %}
                            <i class="bi bi-plus-lg"></i>
                        </button>
                    {% endif %}
                {% endif %}

**Comment by brylie** [comment]
The `activity.id` is a field in the database. The `activity.remaining_eligible_companions` is a method on the `Activity` model. Check `activities/models.py` for the full source code of the `Activity` model.

**Reply by PARKER-X** [reply]
My question is how it is worked data-bs-target="#add-activity-participant-modal-{{ activity.id }}" activity.id is  a single value and 
activity.remaining_eligible_companions gives list of companions how  data is saved using this ?

**Reply by brylie** [reply]
The code above opens a Bootstrap Modal that contains a form.

https://getbootstrap.com/docs/5.0/components/modal/#live-demo 

The data is saved via a form that is submitted by a POST request.

https://github.com/CompanionshipCare/companionship-care/blob/66f33b7926feaac897513db2407eae006c181a1f/project/circles/templates/circles/circle_activity.html#L225
